src/cpp/lpnamer/problem_modifier/ProblemModifier.cpp

```diff
@@ -206,65 +192,83 @@ void ProblemModifier::add_direct_profile_column_constraint(
     std::vector<char> &rowtype, std::vector<double> &rhs,
     std::vector<int> &rstart, const ActiveLink &link,
     const ColumnToChange &column) {
+  unsigned int chronicle_to_use = chronicleToUse(link);
+
   double already_installed_capacity(link.get_already_installed_capacity());
   double direct_already_installed_profile_at_timestep =
-      link.already_installed_direct_profile(column.time_step);
+      link.already_installed_direct_profile(chronicle_to_use, column.time_step);
   rhs.push_back(already_installed_capacity *
                 direct_already_installed_profile_at_timestep);
 
   rowtype.push_back('L');
   colind.push_back(column.id);
   dmatval.push_back(1);
   for (const auto &candidate : link.getCandidates()) {
-    if (candidate.direct_profile(column.time_step) != 0.0) {
+    if (candidateContributionIsNotNull(column, chronicle_to_use, candidate)) {
       colind.push_back(_candidate_col_id[candidate.get_name()]);
-      dmatval.push_back(-candidate.direct_profile(column.time_step));
+      dmatval.push_back(-candidate.directCapacityFactor(chronicle_to_use, column.time_step));
     }
   }
   rstart.push_back((int)dmatval.size());
 }
+bool ProblemModifier::candidateContributionIsNotNull(const ColumnToChange &column,
+                                             unsigned int chronicle_to_use,
+                                             const Candidate &candidate) const {
+  return candidate.directCapacityFactor(chronicle_to_use, column.time_step) != 0.0;
+}
 
 void ProblemModifier::add_indirect_profile_ntc_column_constraint(
     std::vector<double> &dmatval, std::vector<int> &colind,
     std::vector<char> &rowtype, std::vector<double> &rhs,
     std::vector<int> &rstart, const ActiveLink &link,
     const ColumnToChange &column) {
+  unsigned int chronicle_to_use = chronicleToUse(link);
+
   double already_installed_capacity(link.get_already_installed_capacity());
   double indirect_already_installed_profile_at_timestep =
-      link.already_installed_indirect_profile(column.time_step);
+      link.already_installed_indirect_profile(chronicle_to_use, column.time_step);
   rhs.push_back(-already_installed_capacity *
                 indirect_already_installed_profile_at_timestep);
 
   rowtype.push_back('G');
   colind.push_back(column.id);
   dmatval.push_back(1);
   for (const auto &candidate : link.getCandidates()) {
-    if (candidate.indirect_profile(column.time_step) != 0.0) {
+    if (candidate.indirectCapacityFactor(chronicle_to_use, column.time_step) != 0.0) {
```

Comment:
You use a dedicated method `candidateContributionIsNotNull` above, why not here ?

Reply:
This part is now extracted in a new function _candidateContributionIndirectIsNotNull_